[Bug] floating window exceptions: operation was cancelled

## Describe the bug
floating window exceptions wont open after creating a new exception

```
[ERROR] failed to read response from floating exceptions dialog: Gio.IOErrorEnum: Operation was cancelled
```

## To Reproduce

Steps to reproduce the behavior:

1. open a window such as nautilus
2. open the floating window exceptions dialog
3. press the plus button in the top left corner of the exceptions dialog
4. click on nautilus
5. observe the floating window exceptions dialogue has disappeared
6. try to reopen
7. observe logs

## Expected behavior
the window shouldnt close and it should also reopen cleanly

--- a/src/extension.ts
+++ b/src/extension.ts
@@ -573,11 +573,12 @@ export class Ext extends Ecs.System<ExtEvent> {
         };
 
         if (!this.exceptions_ipc)
-            this.exceptions_ipc = utils.async_process_ipc([
-                'gjs',
-                '--module',
-                path,
-            ]);
+            this.exceptions_ipc = utils.async_process_ipc(
+                ['gjs', '--module', path],
+                () => {
+                    this.exceptions_ipc = null;
+                }
+            );
 
         if (this.exceptions_ipc) {
             const generator = (stdout: any, res: any) => {
@@ -3037,6 +3038,7 @@ export default class MosaicExtension extends Extension {
             if (ext.exceptions_ipc) {
                 ext.exceptions_ipc.cancellable.cancel();
                 ext.exceptions_ipc.child.force_exit();
+                ext.exceptions_ipc = null;
             }
 
             _hide_skip_taskbar_windows();

--- a/src/utils.ts
+++ b/src/utils.ts
@@ -157,7 +157,10 @@ export type AsyncIPC = {
     cancellable: any;
 };
 
-export function async_process_ipc(argv: Array<string>): AsyncIPC | null {
+export function async_process_ipc(
+    argv: Array<string>,
+    callback?: () => void
+): AsyncIPC | null {
     const {SubprocessLauncher, SubprocessFlags} = Gio;
 
     const launcher = new SubprocessLauncher({
@@ -188,6 +191,7 @@ export function async_process_ipc(argv: Array<string>): AsyncIPC | null {
     child.wait_async(null, (source: any, res: any) => {
         source.wait_finish(res);
         cancellable.cancel();
+        if (callback) callback();
     });
 
     return {child, stdin, stdout, cancellable};

